packages/react/src/components/TextField/TextField.tsx

```diff
@@ -10,30 +11,29 @@ import { NumericFormat, PatternFormat } from 'react-number-format';
 
 import { isNumericFormat, isPatternFormat } from '../../utils';
 import { InputWrapper } from '../_InputWrapper';
-import type { ReadOnlyVariant_ } from '../_InputWrapper';
+import type { ReadOnlyVariant_, CharLimit } from '../_InputWrapper';
 
-export interface TextFieldProps
-  extends Omit<
-    NumericFormatProps | PatternFormatProps,
-    'readOnly' | 'value' | 'defaultValue'
-  > {
+export type TextFieldProps = {
+  charLimit?: CharLimit;
   defaultValue?: string | number;
   formatting?: TextFieldFormatting;
   isValid?: boolean;
   label?: string;
   readOnly?: boolean | ReadOnlyVariant_;
   value?: string;
-}
+  disabled?: boolean;
+  required?: boolean;
+} & HTMLAttributes<HTMLInputElement>;
```

Comment:
Suggest we keep the existing "extend" as long as we are using the `react-number-format` library as `HTMLAttributes<*>` (and `InputHTMLAttributes<*>` throws error with `react-number-format`) does not include the correct `input` attributes. 

They are however included in the `NumericFormatProps | PatternFormatProps`. 

We can have a look at cleaning this up later :) 

You can still use `type` just replace `HTMLAttributes<HTMLInputElement>` with ```Omit<NumericFormatProps | PatternFormatProps,'readOnly' | 'value' | 'defaultValue'>```